core/http-auth-aws-crt/src/main/java/software/amazon/awssdk/http/auth/aws/crt/AwsCrtV4aHttpSigner.java

```diff
@@ -27,6 +30,73 @@
 @SdkPublicApi
 public interface AwsCrtV4aHttpSigner extends HttpSigner<AwsCredentialsIdentity> {
 
+    /**
+     * The AWS region name to be used for computing the signature.
+     * This property is required.
+     */
+    SignerProperty<String> REGION_NAME =
+        SignerProperty.create(String.class, "RegionName");
+
+    /**
+     * The name of the AWS service.
+     * This property is required.
+     */
+    SignerProperty<String> SERVICE_SIGNING_NAME =
+        SignerProperty.create(String.class, "ServiceSigningName");
+
+    /**
+     * A {@link Clock} to be used at the time of signing.
+     * This property defaults to the time at which signing occurs.
+     */
+    SignerProperty<Clock> SIGNING_CLOCK =
+        SignerProperty.create(Clock.class, "SigningClock");
+
+    /**
+     * The name of the header for the checksum.
+     * This property is optional.
+     */
+    SignerProperty<String> CHECKSUM_HEADER_NAME =
+        SignerProperty.create(String.class, "ChecksumHeaderName");
+
+    /**
+     * The {@link ChecksumAlgorithm} used to compute the checksum.
+     * This property is required *if* a checksum-header name is given.
+     */
+    SignerProperty<ChecksumAlgorithm> CHECKSUM_ALGORITHM =
+        SignerProperty.create(ChecksumAlgorithm.class, "ChecksumAlgorithm");
+
+    /**
+     * A boolean to indicate whether to double url-encode the resource path
+     * when constructing the canonical request.
+     * This property defaults to true.
+     */
+    SignerProperty<Boolean> DOUBLE_URL_ENCODE =
+        SignerProperty.create(Boolean.class, "DoubleUrlEncode");
+
+    /**
+     * A boolean to indicate whether the resource path should be "normalized"
+     * according to RFC3986 when constructing the canonical request.
+     * This property defaults to true.
+     */
+    SignerProperty<Boolean> NORMALIZE_PATH =
+        SignerProperty.create(Boolean.class, "NormalizePath");
+
+    /**
+     * Whether to enable chunked encoding or not.
+     * This property defaults to false.
+     * This is only used in the case of an implementation that supports chunked-encoding.
+     */
+    SignerProperty<Boolean> CHUNKED_ENCODING =
+        SignerProperty.create(Boolean.class, "ChunkedEncoding");
```

Comment:
Does this imply the request is already chunked encoded, or that the signer will make it chunked encoded?

Reply:
Either/or. Currently, the only signer that supports chunked-encoding is S3-signer, and it does the chunked-encoding itself.